src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml.cs

```diff
@@ -13,14 +18,145 @@ namespace NuGet.PackageManagement.UI
     /// </summary>
     public partial class PackageManagerTopPanel : UserControl
     {
-        private FilterLabel _selectedFilter;
+        private TabItem _selectedTabItem
+        {
+            get
+            {
+                return tabsPackageManagement.SelectedItem as TabItem;
+            }
+            set
+            {
+                tabsPackageManagement.SelectedItem = value;
+            }
+        }
+        public TabItem _tabConsolidate { get; private set; }
+        public Border _countConsolidateContainer { get; private set; }
+        public TextBlock _countConsolidate { get; private set; }
 
         public PackageManagerTopPanel()
         {
             InitializeComponent();
+        }
+
+        public void CreateAndAddConsolidateTab(bool isSolution)
+        {
+            IsSolution = isSolution;
+            if (IsSolution)
```

Comment:
You can eliminate a lot of indentation by short-circuiting for the opposite condition:
```C#
if (!IsSolution)
{
    return;
}

…
```